kolibri/plugins/coach/assets/src/views/reports/channel-list-page.vue

```diff
@@ -1,59 +1,84 @@
 <template>
 
-  <report-table :caption="$tr('channelList')">
-    <thead slot="thead">
-      <tr>
-        <header-cell :text="$tr('channels')" align="left"/>
-        <header-cell :text="$tr('lastActivity')" align="left"/>
-      </tr>
-    </thead>
-    <tbody slot="tbody">
-      <tr v-for="row in channelList" :key="row.id">
-        <name-cell :kind="CHANNEL" :title="row.title" :link="reportLink(row.id)"/>
-        <activity-cell :date="lastActive[row.id]" />
-      </tr>
-    </tbody>
-  </report-table>
+  <div>
+    <div v-if="showRecentOnly" ref="recentHeader">
+      <h1>{{ $tr('recentTitle') }}</h1>
+      <sub>{{ $tr('recentSubHeading') }}</sub>
+    </div>
 
-</template>
+    <report-table :caption="$tr('channelList')">
+      <thead slot="thead">
+        <tr>
+          <header-cell :text="$tr('channels')" align="left"/>
+          <header-cell :text="$tr('lastActivity')" align="left"/>
+        </tr>
+      </thead>
+      <tbody slot="tbody">
+        <template v-for="channel in channelList">
+          <tr v-if="channelIsVisible(lastActive[channel.id])" :key="channel.id">
+            <name-cell :kind="CHANNEL" :title="channel.title" :link="reportLink(channel.id)"/>
+            <activity-cell :date="lastActive[channel.id]"/>
+          </tr>
+        </template>
+      </tbody>
+    </report-table>
+  </div>
 
+</template>
 
 <script>
 
-  const ContentNodeKinds = require('kolibri.coreVue.vuex.constants').ContentNodeKinds;
-  const PageNames = require('../../constants').PageNames;
+  const { ContentNodeKinds } = require('kolibri.coreVue.vuex.constants');
+  const { PageNames } = require('../../constants');
   const orderBy = require('lodash/orderBy');
+  const differenceInDays = require('date-fns/difference_in_days');
 
   module.exports = {
     name: 'channelList',
     $trNameSpace: 'coachRecentPageChannelList',
     $trs: {
+      recentTitle: 'Recent Activity',
+      recentSubHeading: 'Showing recent activity in past 7 days',
       channels: 'Channels',
       channelList: 'Channel list',
       lastActivity: 'Last active',
     },
+    data() {
+      return {
+        currentDateTime: Date.now(),
```

Comment:
flag to change this to server time using https://github.com/learningequality/kolibri/pull/1397/files